Sources/Public/DotLottie/DotLottieFile.swift

```diff
@@ -0,0 +1,152 @@
+//
+// DotLottie.swift
+// LottieFiles
+//
+// Created by Evandro Harrison Hoffmann on 27/06/2020.
+// Copyright © 2020 LottieFiles. All rights reserved.
+//
+
+import Foundation
+
+// MARK: - DotLottieFile
+
+/// Detailed .lottie file structure
+public final class DotLottieFile {
+
+  // MARK: Lifecycle
+
+  /// Loads `DotLottie` from `URL` containing .lottie file.
+  ///
+  /// - Parameters:
+  ///  - url: URL to .lottie file
+  ///  - Returns: Deserialized `DotLottie`. Optional.
+  init(url: URL) throws {
+    fileUrl = DotLottieUtils.tempDirectoryURL.appendingPathComponent(url.deletingPathExtension().lastPathComponent)
+    guard url.isDotLottie else {
+      throw DotLottieError.invalidFileFormat
+    }
+    try decompress(from: url, to: fileUrl)
+  }
+
+  /// Loads `DotLottie` from `Data` object containing a compressed animation.
+  ///
+  /// - Parameters:
+  ///  - data: Data of .lottie file
+  ///  - filename: Name of .lottie file
+  ///  - Returns: Deserialized `DotLottie`. Optional.
+  init(data: Data, filename: String) throws {
+    fileUrl = DotLottieUtils.tempDirectoryURL.appendingPathComponent(filename.asFilename())
+    try decompress(data: data, to: fileUrl)
+  }
+
+  // MARK: Public
+
+  /// List of `LottieAnimation` in the file
+  public var animations: [LottieAnimation] = []
+
+  /// Image provider for animations
+  public var imageProvider: AnimationImageProvider?
+
+  // MARK: Internal
+
+  /// Manifest.json file loading
+  var manifest: DotLottieManifest? {
+    let path = fileUrl.appendingPathComponent(DotLottieFile.manifestFileName)
+    return try? DotLottieManifest.load(from: path)
```

Comment:
Should we make this `lazy` or something so it's only loaded once, instead of being loaded repeatedly?

Reply:
If not we may want to call this `loadManifest()` to make it clear that it's reading a file from disk each call